### ISSUE TITLE
Allow WFS search channels to set region based on property

### ISSUE THREAD

**Issue by jampukka** [oskariorg/oskari-docs#280]
Currently single fixed value can be configured as the value for `region` for one WFS search channel (https://github.com/oskariorg/oskari-server/blob/2.8.1/service-search-wfs/src/main/java/fi/nls/oskari/search/channel/WFSSearchChannel.java#L112). We should extend this to allow admin to configure a property to be used for extracting `region` from a Feature.

In addition I propose we allow providing simple (possibly even localized in the future) mapping configuration to optionally alter the value. This could be used to for example map municipality codes (`091`) to human readable region names (`Helsinki`).

**Comment by ZakarFin** [comment]
Sounds good. I would make the `091` -> `Helsinki` mapping a bit more generic than just for the search functionality. Maybe some simple code list config for the WFS-layer/vector source at first that could then be extended to use an external file/service for the code values? Possibly some plugin/hook mechanism that enables that kind of extension?

**Comment by jampukka** [comment]
Right, then I think we should split the work in three separate parts
1) Add support for selecting property to use as value for `region` in `WFSSearchChannel`
2) Change `WFSSearchChannel` to use `OskariWFSClient` (extending search channel support to OGC API Features)
3) Add support for codelists / value transformation mapping support to vector layers and `OskariWFSClient`

**Comment by ZakarFin** [comment]
And maybe part 4 as an optional external source for codelists that can also be added later on. In part 3 I would like to keep the value mapping/codelists out of OskariWFSClient and instead make the client aware of the mapping. So getting features from the service would return the actual values and mapping codes to user-friendly values would be done on the UI.

**Comment by ZakarFin** [comment]
There's an action route that is used to fetch coverage data for layers etc. The same route could be used to pass the codelists to frontend for vector sources.

**Comment by jampukka** [comment]
I'll implement `1` soon. Users can get around of `3` and `4` outside of Oskari (doing it at the service level) so that lowers their priority a little.

**Comment by rekjuh** [comment]
@jampukka like the other issue I just commented on: what's the situation with this? You made a PR on 2022 with the mention on this but is there still something you'd like the Oskari team to work on? If you wish to keep this open and get updates, please create a new issue here: https://github.com/oskariorg/oskari-documentation/issues
